packages/core/saved-objects/core-saved-objects-migration-server-internal/src/model/model.ts

```diff
@@ -219,6 +233,48 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
     } else {
       return throwBadResponse(stateP, res);
     }
+  } else if (stateP.controlState === 'WAIT_FOR_MIGRATION_COMPLETION') {
+    const res = resW as ExcludeRetryableEsError<ResponseType<typeof stateP.controlState>>;
+    const indices = res.right;
+    const aliasesRes = getAliases(indices);
+    if (
+      // If this version's migration has already been completed we can proceed
+      Either.isRight(aliasesRes) &&
+      versionMigrationCompleted(stateP.currentAlias, stateP.versionAlias, aliasesRes.right)
+    ) {
+      return {
+        ...stateP,
+        // Skip to 'OUTDATED_DOCUMENTS_SEARCH_OPEN_PIT' so that if a new plugin was
+        // installed / enabled we can transform any old documents and update
+        // the mappings for this plugin's types.
+        controlState: 'OUTDATED_DOCUMENTS_SEARCH_OPEN_PIT',
```

Comment:
I would have expected to go directly to full completion? This feature is meant only for background task nodes on cloud, right? If so, there's no risk of plugin/config mismatch in theory.

But maybe you gave it more thoughts than I did?

Reply:
yeah I was going back and forth on this. It felt like the safest most conservative approach, but I agree this should be impossible. Especially now with customers having millions of saved objects doing extra work will be expensive, so I will change this.